tests/functional/test_patient_access.py

```diff
@@ -5,32 +5,69 @@
 
 from tests.functional.utils.apigee_api import ApigeeDebugApi
 from tests.functional.utils.helper import (
-    generate_random_email_address,
-    get_add_telecom_email_patch_body,
+    generate_random_phone_number,
+    get_add_telecom_phone_patch_body,
 )
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+AUTH_PATIENT = {
+    "access": "patient",
+    "level": "P9",
+    "login_form": {"username": "9912003071"},
+}
+
+AUTH_PATIENT_P5 = {
+    "access": "patient",
+    "level": "P5",
+    "login_form": {"username": "9912003071"},
+}
+
+AUTH_PATIENT_p5 = {
+    "access": "patient",
+    "level": "p5",
+    "login_form": {"username": "9912003071"},
+}
+
+AUTH_PATIENT_P0 = {
+    "access": "patient",
+    "level": "P0",
+    "login_form": {"username": "9912003071"},
+}
+
+TEST_PATIENT_ID = "9912003071"
 
 
-@pytest.mark.asyncio
 class TestUserRestrictedPatientAccess:
-    async def test_patient_access_retrieve_happy_path(self, nhs_login_token_exchange):
-        token = await nhs_login_token_exchange()
+
+    @pytest.mark.nhsd_apim_authorization(AUTH_PATIENT)
+    def test_patient_access_retrieve_happy_path(self, _nhsd_apim_auth_token_data, add_asid_to_testapp):
```

Comment:
Is there a way of getting the token data without calling a private fixture?

Reply:
Would `nhsd_apim_auth_headers` be better?

Reply:
_nhsd_apim_auth_token_data seems to be used in the examples provided https://nhsd-confluence.digital.nhs.uk/display/APM/Test+Utils+2.0%3A+pytest+plugin+for+APIM, although nhsd_apim_auth_headers seems to be more widely used. We need the token to get the role of the user for the NHSD-Session-URID header, not sure if the nhsd_apim_auth_headers would have it, I'll check

Reply:
The nhsd_apim_auth_headers only has the token, which we would need to parse out in order to make the call for getting the role Id, I would leave it as-is for the time being.